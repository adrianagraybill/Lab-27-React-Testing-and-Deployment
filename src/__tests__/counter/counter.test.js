import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../../components/counter/counter.js';
describe('Does counter work', () => {

  it('<Counter />', () => {
    let component = shallow(<Counter />);
    expect(component.find('span').exists()).toBeTruthy();
  });

  it('the plus button increases the count', () => {
    let component = mount(<Counter />);
    let button = component.find('#increment');
    button.simulate('click');
    expect(component.state('count')).toBe(+1);
  });

  it('the minus button decreases the count', () => {
    let component = mount(<Counter />);
    let button = component.find('#decrement');
    button.simulate('click');
    expect(component.state('count')).toBe(-1);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});