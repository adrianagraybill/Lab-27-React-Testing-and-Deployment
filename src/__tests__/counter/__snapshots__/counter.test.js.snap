// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does counter work renders correctly 1`] = `
<div
  id="counter-container"
>
  <button
    id="decrement"
    onClick={[Function]}
  >
    -
  </button>
  <span>
    Count: 
    0
     
  </span>
  <button
    id="increment"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
